Remove the future tutorial, add links to the async-await codelab

kwalrath opened the pull request:
Fixes #1826.

Staged:
https://sz-www-1.firebaseapp.com/tutorials/language/futures
https://sz-www-1.firebaseapp.com/guides/language/language-tour#declaring-async-functions
https://sz-www-1.firebaseapp.com/tutorials
https://sz-www-1.firebaseapp.com/tutorials/language
...

legalcodes commented on src/_guides/language/index.md:
```diff
@@ -40,7 +40,7 @@ These two resources are popular with both beginning Dartisans and experts.
 <ul markdown="1">
-<li markdown="1">[Futures](/tutorials/language/futures)<br>
+<li markdown="1">[Asynchrony: futures, async, await](/codelabs/async-await)<br>
     A beginner's guide to handling asynchronous programming in Dart.
```
The description we have in the codelab is:
`Learn about and practice writing asynchronous code in DartPad!`

Here we're describing it with:
`A beginner's guide to handling asynchronous programming in Dart.`

Maybe having different descriptions for the codelab is a good thing, just thought I'd mention it.

legalcodes replied:
I think `A beginner's guide` leaves out the DartPad / interactivity of the codelab, where other descriptions have captured that. I think `Learn about and practice writing asynchronous code in DartPad!` works here.

legalcodes replied:
![Screen Shot 2019-08-12 at 11 50 33 AM](https://user-images.githubusercontent.com/5550175/62894596-dce31f80-bd01-11e9-86a5-03d3608688d2.png)

Nesting "Asynchrony" beneath "Asynchronous Programming" can be a little jarring, since "Asynchrony" is more abstract than "Asynchronous Programming". 

Another option might be: 

Change the title of the codelab from "Asynchrony: ..." to "Asynchronous programming: ...", which is what we had for the tutorial and what we're currently doing for the streams tutorial. If so, we could make this section look something like this:

```
Asynchronous Programming:
- futures, async, await
- streams
```


(2) Change the link text to "asynchrony codelab" like we are doing below.

kwalrath replied:
> IMHO `Learn about and practice writing asynchronous code in DartPad!` works here.

Exclamation points don't work anywhere in our docs. :) https://developers.google.com/style/exclamation-points

(I think they're more OK in social sharing text. Possibly OK in codelabs. Not OK anywhere else.)

legalcodes commented on src/_guides/language/language-tour.md:
```diff
@@ -4008,14 +4008,14 @@ Future<String> lookUpVersion() async => '1.0.0';
-{% comment %}
-PENDING: add example here
+For an interactive introduction to using futures, `async`, and `await`,
+see the [asynchrony codelab](/codelabs/async-await).
```
If we change the title of the codelab from "Asynchrony..." to "Asynchronous programming" (see above comment) we may also want to reconsider referring to the codelab as the "asynchrony" codelab. I'll wait to hear your thoughts before brainstorming further.

legalcodes commented on src/_guides/libraries/library-tour.md:
```diff
@@ -1386,12 +1386,12 @@ inputStream
--   [Futures and Error Handling](/guides/libraries/futures-error-handling)
--   [Asynchronous Programming: Streams](/tutorials/language/streams)
--   [Creating Streams in Dart](/articles/libraries/creating-streams)
+-   [Asynchrony: futures, async, await](/codelabs/async-await)
```
See above

legalcodes commented on src/_tutorials/index.md:
```diff
@@ -14,19 +14,14 @@ The following tours assume a basic familiarity with the Dart language,
 [language samples](/samples) or the
 [language tour](/guides/language/language-tour).
+Next, learn about futures by following the
+asynchrony codelab: [Asynchrony: futures, async, await](/codelabs/async-await).
```
One suggestion for reducing duplication of "asynchrony" here might be: 

Next, learn about futures in the following codelab: [Asynchrony: futures, async, await](/codelabs/async-await)

kwalrath commented:
I think I've incorporated all your comments, and I've updated the staged version.

The one downside to changing "asynchrony" to "asynchronous programming" is that the sidenav entry goes from 2 lines to 3. We could just skip the first or second part of the name to fix this. Or we could change the name back to "Asynchrony" for just this one case, since the sidenav doesn't always match the titles anyway. What do you think?

legalcodes commented on src/_data/side-nav.yml:
```diff
@@ -92,9 +92,9 @@
-    - title: Futures & async-await
-      permalink: /tutorials/language/futures
-    - title: Using JSON
+    - title: Futures
```
Is `Futures` the intended title for the codelab under the Development section? How about `Futures, async, await`

legalcodes commented:
> I think I've incorporated all your comments, and I've updated the staged version.
> 
> The one downside to changing "asynchrony" to "asynchronous programming" is that the sidenav entry goes from 2 lines to 3. We could just skip the first or second part of the name to fix this. Or we could change the name back to "Asynchrony" for just this one case, since the sidenav doesn't always match the titles anyway. What do you think?

IMHO skipping the first or second part of the name to fix this is a good solution.

kwalrath commented:
ptal. Should be ready to go now.

legalcodes commented:
LGTM!
